test/library/standard/BigInteger/fitsIntoMethod.chpl

```diff
@@ -0,0 +1,156 @@
+// var c_types_to_check: [0..#6] type = [c_short, c_ushort, c_int, c_uint, c_long, c_ulong];
+
+const lower_bounds : [0..#8] string = ["-128", "0", "-32768", "0", "-2147483648", "0", "-9223372036854775808", "0"];
+const upper_bounds : [0..#8] string = ["127", "255", "32767", "65535", "2147483647", "4294967295", "9223372036854775807", "18446744073709551615"];
+
+writeln("Checking Chapel Types:");
+for i in 0..<8 {
+    const b_lower_in = new bigint(lower_bounds[i]);
+    const b_upper_in = new bigint(upper_bounds[i]);
+
+    const b_lower_out = b_lower_in - 1;
+    const b_upper_out = b_upper_in + 1;
+
+    // writeln("in: (", b_lower_in, ", ", b_upper_in, ") \t out: (", b_lower_out, ", ", b_upper_out, ")");
+
+    var correct = true;
+
+    select i {
+        when 0 {
+            correct &= b_lower_in.fitsInto(int(8));
+            correct &= b_upper_in.fitsInto(int(8));
+
+            correct &= !b_lower_out.fitsInto(int(8));
+            correct &= !b_upper_out.fitsInto(int(8));
+        }
+        when 1 {
+            correct &= b_lower_in.fitsInto(uint(8));
+            correct &= b_upper_in.fitsInto(uint(8));
+
+            correct &= !b_lower_out.fitsInto(uint(8));
+            correct &= !b_upper_out.fitsInto(uint(8));
+        }
+        when 2 {
+            correct &= b_lower_in.fitsInto(int(16));
+            correct &= b_upper_in.fitsInto(int(16));
+
+            correct &= !b_lower_out.fitsInto(int(16));
+            correct &= !b_upper_out.fitsInto(int(16));
+        }
+        when 3 {
+            correct &= b_lower_in.fitsInto(uint(16));
+            correct &= b_upper_in.fitsInto(uint(16));
+
+            correct &= !b_lower_out.fitsInto(uint(16));
+            correct &= !b_upper_out.fitsInto(uint(16));
+        }
+        when 4 {
+            correct &= b_lower_in.fitsInto(int(32));
+            correct &= b_upper_in.fitsInto(int(32));
+
+            correct &= !b_lower_out.fitsInto(int(32));
+            correct &= !b_upper_out.fitsInto(int(32));
+        }
+        when 5 {
+            correct &= b_lower_in.fitsInto(uint(32));
+            correct &= b_upper_in.fitsInto(uint(32));
+
+            correct &= !b_lower_out.fitsInto(uint(32));
+            correct &= !b_upper_out.fitsInto(uint(32));
+        }
+        when 6 {
+            correct &= b_lower_in.fitsInto(int(64));
+            correct &= b_upper_in.fitsInto(int(64));
+
+            correct &= !b_lower_out.fitsInto(int(64));
+            correct &= !b_upper_out.fitsInto(int(64));
+        }
+        when 7 {
+            correct &= b_lower_in.fitsInto(uint(64));
+            correct &= b_upper_in.fitsInto(uint(64));
+
+            correct &= !b_lower_out.fitsInto(uint(64));
+            correct &= !b_upper_out.fitsInto(uint(64));
+        }
+        otherwise { writeln("wut?"); }
+    }
+
+    writeln(correct);
+}
+
+writeln("Checking C Types:");
+for i in 0..<8 {
```

Comment:
For this section, I think you're going to run into what I was talking about earlier, where some systems have defined a different size for especially `c_ulong`.  That doesn't mean you shouldn't check this method works there, but it means this test will fail in our nightly linux32 run unless we do something about it.

I think there's a few ways to solve this problem:

- You can have this test file itself check which type is expected to be equivalent depending on the settings
- You might be able to write alternate .good files that can be chosen based on the settings.  We don't have full support for every possible variation, so it might not be viable for linux32, but if it is, you can find general information on that [here](https://chapel-lang.org/docs/latest/developer/bestPractices/TestSystem.html#testing-different-behavior-in-different-settings) and an example of a specific linux32 .good file in the test directory
- You might be able to write a .prediff/.precomp/.preexec file for the test that will generate the appropriate output into a .good file depending on the system.  I don't remember off the top of my head when we need the .good file to already exist in order to check it, so you might need to do some experimentation or use `greptests` to find a test that generates its .good file.
- You could make a .skipif for this test when the platform behaves differently than this test expects, and have an alternate copy of the test that checks the other sizes and uses a skipif to avoid settings where the original test was correct.  E.g.

fitsIntoMethod.skipif:
```
CHPL_TARGET_PLATFORM==linux32
```

fitsIntoMethod.linux32.skipif (to go with `fitsIntoMethod.linux32.chpl` or whatever you want that variant to be called)
```
CHPL_TARGET_PLATFORM==linux64
```

I think either of those four choices would be viable and there may be others I'm not thinking of